### PR TITLE
rtext: added functions for camel case and snake case (reopened due to formatting errors)

### PR DESCRIPTION
**Pull request by IoIxD**
Adds functions for converting text to camel and snake case to match the addition of a pascal case function. Both have been tested against each other's conventions.

This is a duplicate of a PR that was opened earlier. I forgot to turn my auto formatter off so the PR had 6,000 changes. I used git reset, redid the changes, and then forced pushed the commit; as a result, the old PR cannot be merged, so I have to open a new one.

And if I am wrong, then this is what I am seeing on my screen, which is of importance:

image
image
image

### CODE REVIEW

File: src/rtext.c [hunk 1]
```diff
@@ -1786,6 +1786,78 @@
...
+        for (int i = 1, j = 1; (i < MAX_TEXT_BUFFER_LENGTH - 1) && (text[j] != '\0'); i++, j++)
+        {
+            if (text[j] != '_')
+                buffer[i] = text[j];
+            else
+            {
+                j++;
+                if ((text[j] >= 'a') && (text[j] <= 'z'))
+                {
+                    if (i == 0)
```
[the remark this answers is not in the document]
Comment (IoIxD) [reply src/rtext.c]: Done.

**Comment by IoIxD** [comment]
Forgot to commsnt btw but the issue is fixed.
The other issues GitHub found are present in TextToPascal and not relevant.

**Comment by raysan5** [comment]
@IoIxD Thanks for the addition! Probably not many users would need that functionality but I personally could need it.

**Comment by IoIxD** [comment]
> @IoIxD Thanks for the addition! Probably not many users would need that functionality but I personally could need it.

No problem! I was also actually just thibking of a fourth function that could turn any of these cases into properly spaced/capitalized case (snake_case -> Snake Case). Would you be down for that?

**Comment by raysan5** [comment]
> Would you be down for that?

@IoIxD I think it's ok for now, the provided functions already cover most of the possible use cases for this kind of functions. Personally I find it useful when generating code from data (like in parsers and similars...).

**Pull request merged**